codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -303,27 +381,34 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 
 	var reconcileEvent {{.reconcilerEvent|raw}}
 	if resource.GetDeletionTimestamp().IsZero() {
-		// Append the target method to the logger.
-		logger = logger.With(zap.String("targetMethod", "ReconcileKind"))
-
-		// Set and update the finalizer on resource if r.reconciler
-		// implements Finalizer.
-		if resource, err = r.setFinalizerIfFinalizer(ctx, resource); err != nil {
-			return {{.fmtErrorf|raw}}("failed to set finalizers: %w", err)
-		}
+		if isLeader {
+			// Append the target method to the logger.
+			logger = logger.With(zap.String("targetMethod", "ReconcileKind"))
+
+			// Set and update the finalizer on resource if r.reconciler
+			// implements Finalizer.
+			if resource, err = r.setFinalizerIfFinalizer(ctx, resource); err != nil {
+				return {{.fmtErrorf|raw}}("failed to set finalizers: %w", err)
+			}
+			{{if .isKRShaped}}
+			reconciler.PreProcessReconcile(ctx, resource)
+			{{end}}
 
-		{{if .isKRShaped}}
-		reconciler.PreProcessReconcile(ctx, resource)
-		{{end}}
+			// Reconcile this copy of the resource and then write back any status
+			// updates regardless of whether the reconciliation errored out.
+			reconcileEvent = r.reconciler.ReconcileKind(ctx, resource)
 
-		// Reconcile this copy of the resource and then write back any status
-		// updates regardless of whether the reconciliation errored out.
-		reconcileEvent = r.reconciler.ReconcileKind(ctx, resource)
+			{{if .isKRShaped}}
+			reconciler.PostProcessReconcile(ctx, resource, original)
```

Comment:
looks like the krshaped pre/post process are properly preserved! Thanks for  the merge resolution

Reply:
/lgtm

Reply:
cool thanks